crypto/bytestring/cbs.c

```diff
@@ -477,12 +477,15 @@ int CBS_get_asn1_uint64(CBS *cbs, uint64_t *out) {
 
 // Adding warning suppression as temporary fix for gcc11 ARM build issue
 // https://github.com/awslabs/aws-lc/issues/184
-#if defined(__has_warning)
-#  if  __has_warning("-Wstringop-overflow")
-#    pragma GCC diagnostic push
-#    pragma GCC diagnostic ignored "-Wstringop-overflow"
-#  endif
-#endif
+// First two warnings are turned on to avoid incompatibilites with older compilers.
+#if defined(__GNUC__)
+
+# pragma GCC diagnostic push
+# pragma GCC diagnostic ignored "-Wpragmas"
+# pragma GCC diagnostic ignored "-Wunknown-warning-option"
+# pragma GCC diagnostic ignored "-Wstringop-overflow"
+
+#endif // __GNUC__
```

Comment:
Nit: do we want to make the condition more specific by including the version check? e.g.
```
defined(__GNUC__) && __GNUC__ >= ? // gcc x ?
```

Reply:
Oh I didn't realize that existed, I'll add that in there 👍